addons/sale/tests/test_sale_pricelist.py

```diff
@@ -161,3 +161,50 @@ def test_sale_with_pricelist_discount_excluded(self):
                 else:  # no discount for the rest
                     self.assertEqual(line.discount, 0.0, 'Pricelist of SO should not be applied on an order line')
                     self.assertEqual(line.price_unit, line.product_id.list_price, 'Unit price of order line should be a sale price as the pricelist not applied on the other category\'s product')
+
+    def test_sale_change_of_pricelists_excluded_value_discount(self):
+        """ Test SO with the pricelist 'discount displayed' and check displayed percentage value after multiple changes of pricelist """
+
+        # Create a pricelist without discount policy: percentage on all products
+        pricelist_discount_excl_global = self.env['product.pricelist'].create({
+            'name': 'Pricelist C',
+            'discount_policy': 'without_discount',
+            'company_id': self.env.company.id,
+        })
+        self.env['product.pricelist.item'].create({
+            'pricelist_id': pricelist_discount_excl_global.id,
+            'applied_on': '3_global',
+            'compute_price': 'percentage',
+            'percent_price': 54,
+        })
+
+        # Create a product with a very low price
+        amazing_product = self.env['product.product'].create({
+            'name': 'Amazing Product',
+            'lst_price': 0.03,
+        })
+
+        # create a simple Sale Order with a unique line
+        sale_order = self.env['sale.order'].create({
+            'partner_id': self.partner_a.id,
+            'partner_invoice_id': self.partner_a.id,
+            'partner_shipping_id': self.partner_a.id,
+            'pricelist_id': self.company_data['default_pricelist'].id,
+        })
+        self.one_product_order = self.env['sale.order.line'].create({
+            'name': amazing_product.name,
+            'product_id': amazing_product.id,
+            'product_uom_qty': 1,
+            'product_uom': amazing_product.uom_id.id,
+            'price_unit': 0.03,
+            'order_id': sale_order.id,
+            'tax_id': False,
+        })
+
+        # Change the pricelist
+        sale_order.write({'pricelist_id': pricelist_discount_excl_global.id})
+        # Update Prices
+        sale_order.update_prices()
+
+        # Check that the discount displayed is the correct one
+        self.assertEqual(sale_order.order_line.discount, 54, 'Discount for product %s should be 54 percent with pricelist %s' % (sale_order.order_line.name, pricelist_discount_excl_global.name))
```

Comment:
Split the line to respect the limit of 100 chars, and keep the explanation short, the product & pricelist names are not really useful, you will have the stack trace in the traceback, so you know which test is failing anyway.
```suggestion
        self.assertEqual(
            sale_order.order_line.discount, 54,
            "Wrong discount computed for specified product & pricelist"
        )
```